src/authenticationservices.cs

```diff
@@ -192,5 +227,481 @@ interface ASWebAuthenticationSession {
 
 		[Export ("cancel")]
 		void Cancel ();
+
+		[iOS (13,0)]
+		[NullAllowed, Export ("presentationContextProvider", ArgumentSemantic.Weak)]
+		ASWebAuthenticationPresentationContextProviding PresentationContextProvider { get; set; }
+
+		[iOS (13,0)]
+		[Export ("prefersEphemeralWebBrowserSession")]
+		bool PrefersEphemeralWebBrowserSession { get; set; }
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorization {
+
+		[Export ("provider", ArgumentSemantic.Strong)]
+		IASAuthorizationProvider Provider { get; }
+
+		[Export ("credential", ArgumentSemantic.Strong)]
+		IASAuthorizationCredential Credential { get; }
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	enum ASAuthorizationScope {
+		[Field ("ASAuthorizationScopeFullName")]
+		FullName,
+		[Field ("ASAuthorizationScopeEmail")]
+		Email,
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[Native]
+	enum ASUserDetectionStatus : long {
+		Unsupported,
+		Unknown,
+		LikelyReal,
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject), Name = "ASAuthorizationAppleIDCredential")]
+	[DisableDefaultCtor]
+	interface ASAuthorizationAppleIdCredential : ASAuthorizationCredential {
+
+		[Export ("user")]
+		string User { get; }
+
+		[NullAllowed, Export ("state")]
+		string State { get; }
+
+		[Export ("authorizedScopes", ArgumentSemantic.Copy)]
+		[BindAs (typeof (ASAuthorizationScope []))]
+		NSString[] AuthorizedScopes { get; }
+
+		[NullAllowed, Export ("authorizationCode", ArgumentSemantic.Copy)]
+		NSData AuthorizationCode { get; }
+
+		[NullAllowed, Export ("identityToken", ArgumentSemantic.Copy)]
+		NSData IdentityToken { get; }
+
+		[NullAllowed, Export ("email")]
+		string Email { get; }
+
+		[NullAllowed, Export ("fullName", ArgumentSemantic.Copy)]
+		NSPersonNameComponents FullName { get; }
+
+		[Export ("realUserStatus")]
+		ASUserDetectionStatus RealUserStatus { get; }
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[Native]
+	enum ASAuthorizationAppleIdProviderCredentialState : long {
+		Revoked,
+		Authorized,
+		NotFound,
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject), Name = "ASAuthorizationAppleIDProvider")]
+	[DisableDefaultCtor] // NSInvalidArgumentException Reason: -[ASAuthorizationAppleIDRequest init]: unrecognized selector sent to instance 0x600000950000
```

Comment:
How are you supposed to create/get an instance of this type then? From the code sample here, it seems you're supposed to use the default constructor: https://developer.apple.com/documentation/authenticationservices/asauthorizationappleidprovider?language=objc

Reply:
```Objective-C exception thrown.  Name: NSInvalidArgumentException Reason: -[ASAuthorizationAppleIDRequest init]: unrecognized selector sent to instance 0x600001421770
Native stack trace:
	0   CoreFoundation                      0x0000000102e3422e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00000001101d4f57 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000102e547f4 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   AuthenticationServices              0x000000010ffbe7b5 -[ASAuthorizationRequest init] + 25
	4   introspection                       0x00000001026a3f59 xamarin_dyn_objc_msgSend + 217
	5   ???                                 0x000000013e4b8902 0x0 + 5340104962
```
So it's not `init` itself but something it calls... we might be missing some configuration steps

Reply:
oops, was simply copy/pasted on the wrong line (comment is fine)